From 718e4b3d3daa81f7d1cba3337b5deaf404a5cc6c Mon Sep 17 00:00:00 2001
From: Vinod Koul <vinod.koul@intel.com>
Date: Fri, 5 Dec 2014 11:20:44 +0530
Subject: [PATCH 01/39] ASoC: codecs: fix compilation breakage!!!

The wolfson backport was merged recently but looks like no compile tests
were done. The backport merge was broken as commit a0259c2e move soc_enum
variable max to items, ideally this change should have been committed as
series of changes from upstream backports, but...
so fix it now here

Change-Id: Ib68abd8fbfb704a15a424f02e98d784de2435cec
Signed-off-by: Vinod Koul <vinod.koul@intel.com>
---
 sound/soc/codecs/arizona.c | 2 +-
 sound/soc/codecs/wm_adsp.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/sound/soc/codecs/arizona.c b/sound/soc/codecs/arizona.c
index a3d37a4..9a27497 100644
--- a/sound/soc/codecs/arizona.c
+++ b/sound/soc/codecs/arizona.c
@@ -769,7 +769,7 @@ int arizona_put_anc_input(struct snd_kcontrol *kcontrol,
 	int sel = ucontrol->value.enumerated.item[0];
 	unsigned int val, mask, shift;
 
-	if (sel >= e->max)
+	if (sel >= e->items)
 		return -EINVAL;
 
 	switch (e->reg) {
diff --git a/sound/soc/codecs/wm_adsp.c b/sound/soc/codecs/wm_adsp.c
index 8908081..6c5d52e 100644
--- a/sound/soc/codecs/wm_adsp.c
+++ b/sound/soc/codecs/wm_adsp.c
@@ -2406,7 +2406,7 @@ int wm_adsp2_init(struct wm_adsp *adsp, struct mutex *fw_lock)
 		for (i = 0; i < adsp->num_firmwares; i++)
 			ctl_names[i] = adsp->firmwares[i].name;
 
-		wm_adsp_fw_enum[adsp->num - 1].max = adsp->num_firmwares;
+		wm_adsp_fw_enum[adsp->num - 1].items = adsp->num_firmwares;
 		wm_adsp_fw_enum[adsp->num - 1].texts = ctl_names;
 	}
 
-- 
1.9.1

